docs/src/references/c8y-log-management.md

```diff
@@ -0,0 +1,122 @@
+# Log file management from Cumulocity
+
+Thin-edge provides an operation plugin to [fetch log files on the device from Cumulocity](https://cumulocity.com/guides/users-guide/device-management/#logs).
+
+* Log file management from Cumulocity is provided with a `c8y_log_plugin` which runs as a daemon on thin-edge.
+* The device owner can define the list of log files that can be retrieved from Cumulocity,
+  in the plugin's configuration file named `c8y_log_plugin.toml`.
+* On receipt of a log file request for a given type, the log file for that type is retrieved from this `c8y_log_plugin.toml`.
+* The list of managed log files can be updated both locally as well as from Cumulocity cloud.
+* This plugin's configuration file can be managed from the cloud using the configuration management feature of Cumulocity,
+  combined with the `c8y_configuration_plugin` of thin-edge.
+
+## Installation
+
+As part of this plugin installation:
+* An empty file is created at `/etc/tedge/operations/c8y/c8y_LogfileRequest` indicating that this device supports `c8y_LogfileRequest` operation from Cumulocity.
+* On systemd enabled devices, the service definition file for this `c8y_log_plugin` daemon is also installed as part of this plugin installation.
+
+Once installed, the `c8y_log_plugin` is run as a daemon on the device listening to log requests from Cumulocity on `c8y/s/us` MQTT topic.
+On startup, it reports all the log file types that it manages, defined in the `c8y_log_plugin.toml`
+
+## Configuration
+
+The `c8y_log_plugin` configuration is stored by default under `/etc/tedge/c8y/c8y_log_plugin.toml`.
+
+This [TOML](https://toml.io/en/) file defines the list of log files that can be retrieved from the cloud tenant.
+The paths to these files can be represented using [glob](https://en.wikipedia.org/wiki/Glob_(programming)) patterns.
+The `type` given to these paths are used as the log type when they are reported to Cumulocity.
```

Comment:
I would add that the path pattern is used a the default type name is the latter is not provided.

Reply:
I intentionally avoided that contract as the glob patterns of the paths can contain special characters like `*`. When log upload events are created, this `type` needs to be used as the Cumulocity event's `type` as well. I'm not sure if using special characters is allowed in that case.

I guess I should have documented that Cumulocity contract with `type` explicitly. Will do so.